### PR TITLE
SSO Service Name Migration

### PR DESCRIPTION
**Pull request by ShellyXueHan**
Files to review:
- SSO community notification at `docs/kcMigration/sso-migration-notification.md `

### CODE REVIEW

File: docs/kcMigration/sso-migration-notification.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+
+### What is happening?
+
+Hi team, the KeyCloak SSO service is finally removing its *Pathfinder* sticker and getting a new name.  This change does not affect the service offering but does change how you will access it.  The name is changing from sso.pathfinder.gov.bc.ca to oidc.gov.bc.ca.  For more information on this Single Sign-On service, refer to [here](https://developer.gov.bc.ca/BC-Government-SSO-Service-Definition).
```
Comment (jefkel) [inline docs/kcMigration/sso-migration-notification.md:4]:
```suggestion
Hi team, the KeyCloak SSO service is finally starting the process of removing its *Pathfinder* sticker and getting a new name.  This change does not affect the service offering but will eventually change how you will access it.  The name is changing from sso.pathfinder.gov.bc.ca to oidc.gov.bc.ca.  For more information on this Single Sign-On service, refer to [here](https://developer.gov.bc.ca/BC-Government-SSO-Service-Definition).
```

File: docs/kcMigration/sso-migration-notification.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+1 week of time are set aside for Test environment testing before the Production environment service name change.
+
+#### Production SSO New Service Name Enable on `Oct 8th`
+Current SSO service disruption for the new production service name is scheduled for 5:30PM - 7:30PM (2 hours), on **October 8th, 2020**.
+
+
+### Will there be an impact on the Platform apps?
+
+Yes there will be service disruption when we enable the new SSO service name, but the effects are expected to be temporary and will last for about 10mins for each realm. This is caused by KeyCloak internal settings update from Platform Services team, if you are interested in more details, refer to [doc](./KeycloakMigration.md).
```
Comment (jefkel) [inline docs/kcMigration/sso-migration-notification.md:38]: is this in addition to the 1.5 - 2 hour service disruptions mentioned above?

File: docs/kcMigration/sso-migration-notification.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+Also note that production SSO service disruption will be scheduled outside of business hours, after 5:00PM.
+
+Once the service name enablment is complete, SSO service will return back to normal. What you should be expecting is that KeyCloak would then be accessible via both service names, `sso-*.pathfinder.gov.bc.ca` and `*.oidc.gov.bc.ca`.
+
+
+### Do I need to do anything?
+
+Yes, you will need to participate in both `Step 2 & 3` after new service name is enabled.
```
Comment (jefkel) [inline docs/kcMigration/sso-migration-notification.md:47]: Cleaning up after removing step 3
```suggestion
```

File: docs/kcMigration/sso-migration-notification.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
```
Comment (jefkel) [inline docs/kcMigration/sso-migration-notification.md:48]:
```suggestion
```

File: docs/kcMigration/sso-migration-notification.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+For Step 2, you should check on all existing usages of SSO KeyCloak immediately. There will be __one week__ to report any bugs or integration issues, before we move onto the next environment. Platform Service team will able available to help troubleshoot issues relevant to the service name change when any SSO admin action is needed.
```
Comment (jefkel) [inline docs/kcMigration/sso-migration-notification.md:49]:
```suggestion
You should check on all existing usages of SSO KeyCloak after the new name has been added to confirm your use of the sso-dev.pathfinder.gov.bc.ca service name is not impacted. There will be __one week__ to report any bugs or integration issues, before we move onto the next environment. Platform Service team will able available to help troubleshoot issues relevant to the service name change when any SSO admin action is needed.
```

File: docs/kcMigration/sso-migration-notification.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+For Step 3, you should start to update the SSO service name for all application integrations and test out the authentication flow as soon as possible. (If you have BCSC integration with your realm, stay tune for more details on how to proceed with the update. We will be working with some polit teams and BCSC team once Dev SSO has the new service name in place.)
```
Comment (jefkel) [inline docs/kcMigration/sso-migration-notification.md:51]:
```suggestion
Once the change has been completed in an environment, you can begin to update the SSO service name for all application integrations and test out the authentication flow with the new enterprise service name. (If you have BCSC integration with your realm, stay tuned for more details on how to proceed with the update. We will be working with some polit teams and BCSC team once Dev SSO has the new service name in place.)
```

**Review by jefkel — CHANGES_REQUESTED** [review]
overall, it read like we are pushing to remove the pathfinder name within the next 3 weeks.  I'd recommend keeping the focus on the immediate change (adding a new name), and then we can begin pushing the teams to test out the new name after the change is completed.

File: docs/kcMigration/sso-migration-notification.md (hunk 2 above)
Comment (ShellyXueHan) [reply docs/kcMigration/sso-migration-notification.md:38]: no it's included. the 1.5 to 2 hours includes the 10 mins disruption for each realm.

**Comment by ShellyXueHan** [comment]
> overall, it read like we are pushing to remove the pathfinder name within the next 3 weeks. I'd recommend keeping the focus on the immediate change (adding a new name), and then we can begin pushing the teams to test out the new name after the change is completed.

Good point, I like it to separate out the two phases and focus on the just communicating the new service name for now. I'll update some titles and terms as well.

**Review by mitovskaol — APPROVED** [review]

**Review by jefkel — APPROVED** [review]
looks great

File: docs/kcMigration/sso-migration-notification.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+Once the service name enablment is complete, SSO service will return back to normal. What you should be expecting is that KeyCloak would then be accessible via both service names, `sso-*.pathfinder.gov.bc.ca` and `*.oidc.gov.bc.ca`.
+
+
+### Do I need to do anything?
+
+You should check on all existing usages of SSO KeyCloak after the new name has been added to confirm your use of the sso-dev.pathfinder.gov.bc.ca service name is not impacted. There will be __one week__ to report any bugs or integration issues, before we move onto the next environment. Platform Service team will able available to help troubleshoot issues relevant to the service name change when any SSO admin action is needed.
+
+Once the change has been completed in an environment, you can begin to update the SSO service name for all application integrations and test out the authentication flow with the new enterprise service name. (If you have BCSC integration with your realm, stay tuned for more details on how to proceed with the update. We will be working with some polit teams and BCSC team once Dev SSO has the new service name in place.)
+
+Here are the recommended ways to start with:
```
Comment (patricksimonian) [inline docs/kcMigration/sso-migration-notification.md:49]:
```suggestion
### Here are the recommended ways to start with:
```

**Review by patricksimonian — APPROVED** [review]

**Pull request merged**